fdbserver/VersionedBTree.actor.cpp

```diff
@@ -954,6 +958,19 @@ class FIFOQueue {
 				if (c.pageID == invalidLogicalPageID || c.pageID == c.endPageID) {
 					debug_printf("FIFOQueue::Cursor(%s) peekAllExt Queue exhausted\n", c.toString().c_str());
 					res.send(results);
+
+					// Since we have reached the end of the queue, verify that the number of entries read matches
+					// the queue metadata. If it does, send end_of_stream() to mark completion, else throw an error
+					entriesRead += results.size();
```

Comment:
This addition is done in both the end-of-queue case and end-of-extent case, but why not simply increment `entriesRead` right next to the single place where `results.push_back()` is done?

Reply:
Oh just because it will be updated far fewer times this way (once per extent vs once per entry). Do you prefer updating per entry?

Reply:
Having two increments is fine, but then I think the variable should be called `entriesSent` or `entriesWritten` instead of `entriesRead` because most of the time it doesn't accurately represent how many entries have been read from the queue pages.

For the name `entriesRead` I think it should be incremented once, at the point where the entry is _read_ and pushed onto the result buffer.  I'm not terribly concerned about performance - the push work probably dwarfs it and `entriesRead` will be adjacent to `results` in the generated `ActorState` which is also changing.

Reply:
Yeah I guess, shouldn't worry about performance here. I changed it to the latter.